pkg/controllers/controllers.go

```diff
@@ -99,25 +96,16 @@ func NewControllers(
 		nodeclaimdisruption.NewController(clock, kubeClient, cloudProvider),
 		nodeclaimhydration.NewController(kubeClient, cloudProvider),
 		nodehydration.NewController(kubeClient, cloudProvider),
-		status.NewController[*v1.NodeClaim](
-			kubeClient,
-			mgr.GetEventRecorderFor("karpenter"),
-			status.EmitDeprecatedMetrics,
-			status.WithHistogramBuckets(prometheus.ExponentialBuckets(0.5, 2, 15)), // 0.5, 1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1024, 2048, 4096, 8192
-			status.WithLabels(append(lo.Map(cloudProvider.GetSupportedNodeClasses(), func(obj status.Object, _ int) string { return v1.NodeClassLabelKey(object.GVK(obj).GroupKind()) }), v1.NodePoolLabelKey)...),
-		),
-		status.NewController[*v1.NodePool](
-			kubeClient,
-			mgr.GetEventRecorderFor("karpenter"),
-			status.EmitDeprecatedMetrics,
-			status.WithHistogramBuckets(prometheus.ExponentialBuckets(0.5, 2, 15)), // 0.5, 1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1024, 2048, 4096, 8192
-		),
-		status.NewGenericObjectController[*corev1.Node](
-			kubeClient,
-			mgr.GetEventRecorderFor("karpenter"),
-			status.WithHistogramBuckets(prometheus.ExponentialBuckets(0.5, 2, 15)), // 0.5, 1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1024, 2048, 4096, 8192
-			status.WithLabels(append(lo.Map(cloudProvider.GetSupportedNodeClasses(), func(obj status.Object, _ int) string { return v1.NodeClassLabelKey(object.GVK(obj).GroupKind()) }), v1.NodePoolLabelKey, v1.NodeInitializedLabelKey)...),
-		),
+	}
+
+	if !options.FromContext(ctx).FeatureGates.DisableMetricsControllers {
+		controllers = append(controllers, metricspod.NewController(kubeClient, cluster),
+			metricsnodepool.NewController(kubeClient, cloudProvider),
+			metricsnode.NewController(cluster),
+			status.NewController[*v1.NodeClaim](kubeClient, mgr.GetEventRecorderFor("karpenter"), status.EmitDeprecatedMetrics, status.WithLabels(append(lo.Map(cloudProvider.GetSupportedNodeClasses(), func(obj status.Object, _ int) string { return v1.NodeClassLabelKey(object.GVK(obj).GroupKind()) }), v1.NodePoolLabelKey)...)),
```

Comment:
Is there a reason for removal of custom buckets?

Reply:
+1